code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -191,26 +191,27 @@
 	origin_tech = null
 	selfcharge = 1
 	can_charge = 0
-	var/inawaymission = 1
+	// Selfcharge is enabled and disabled, and used as the away mission tracker
+	selfcharge = TRUE
 
-/obj/item/gun/energy/laser/awaymission_aeg/process()
-	var/turf/my_loc = get_turf(src)
-	if(is_away_level(my_loc.z))
-		if(inawaymission)
-			return ..()
+/obj/item/gun/energy/laser/awaymission_aeg/Initialize(mapload)
+	. = ..()
+	// Force update it incase it spawns outside an away mission and shouldnt be charged
+	onTransitZ(new_z=loc.z)
```

Comment:
spacing.

Reply:
I thought args for procs didnt have spaces. Huh.

Reply:
`onTransitZ(new_z = loc.z)` will work.

Reply:
I meant in our code style. I will change it now I can switch branches.